src/vs/workbench/electron-sandbox/window.ts

```diff
@@ -697,23 +697,44 @@ export class NativeWindow extends Disposable {
 			}
 		}
 
-		// Windows 7 warning
+		// Windows 7/8/8.1 warning
 		if (isWindows) {
 			const version = this.environmentService.os.release.split('.');
+			const majorVersion = version[0];
+			const minorVersion = version[1];
+			const eolReleases = new Map<string, Map<string, string>>([
+				['6', new Map<string, string>([
+					['1', 'Windows 7 / Windows Server 2008 R2'],
+					['2', 'Windows 8 / Windows Server 2012'],
+					['3', 'Windows 8.1 / Windows Server 2012 R2'],
+				])],
+			]);
```

Comment:
Should this be extracted outside of this method?

Reply:
Version mappings are different between Windows and MacOS, so keeping them separate is better for simplicity.